core/src/consensus/stake/action_data.rs

```diff
@@ -184,6 +188,32 @@ impl<'a> Delegation<'a> {
     }
 }
 
+#[derive(Default, RlpEncodable, RlpDecodable, Debug, PartialEq)]
+pub struct Metadata {
+    pub block_number: u64,
```

Comment:
I think this `Metadata` should move to the `TopLevelState`. `Term` can be used outside of the stake custom handler.

Reply:
Fixed.